Add ability to decode signet

thebrandonlucas opened the pull request:
### Describe the changes you have made in this PR

This PR begins the process of properly decoding signet payments with Alby as mentioned by #3213. The issue is Alby doesn't seem to have signet enabled anywhere that I could find in the app, and I'm not familiar enough with the codebase to know where to conditionally retrieve the network. The [bolt11]() library Alby is using to decode invoices takes an optional second parameter that allows specification of a custom network, i.e.:

```
  const signet = {
    bech32: "tbs",
    pubKeyHash: 0x6f,
    scriptHash: 0xc4,
    validWitnessVersions: [0],
  };
  const paymentRequestDetails = lightningPayReq.decode(paymentRequest, signet);
```

Right now, the changes aren't very DRY and make payments only work with signet, so if someone more knowledgeable could show me how to grab the network dynamically and place the signet constant in the appropriate file, I would be happy to finish implementing.


### Link this PR to an issue [optional]

Fixes _#3213_

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]
Note that this is the signet Mutiny app

https://github.com/getAlby/lightning-browser-extension/assets/38222767/809f6042-fae3-458e-8d05-73075dfbb055


### How has this been tested?


### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [ ] Added automated tests where applicable
- [ ] Update Docs & Guides
- For UI-related changes
- [ ] Darkmode
- [ ] Responsive layout

rolznz commented:
@thebrandonlucas awesome, thanks for looking into this!

Actually, ideally the bolt11 package would actually support mutinynet. It already supports testnet/mainnet so it could also do this check internally. What do you think?

thebrandonlucas commented:
@rolznz I've updated the code and taking it out of draft

rolznz commented:
Hi @thebrandonlucas 

After looking at your changes, your first suggestion (forking the bolt11 library and using that until the PR gets merged) seems like the best option to me, as adding mutinynet there would be a very minor change and more consistent (it's the same checks as it already does for e.g. regtest). Then we just change the dependency in our package.json.

thebrandonlucas commented:
> After looking at your changes, your first suggestion (forking the bolt11 library and using that until the PR gets merged) seems like the best option to me, as adding mutinynet there would be a very minor change and more consistent (it's the same checks as it already does for e.g. regtest). Then we just change the dependency in our package.json.

@rolznz Done. I modified the bolt11 library to [bolt11-signet](https://www.npmjs.com/package/bolt11-signet), opened a [PR](https://github.com/bitcoinjs/bolt11/pull/73) to the original bolt11 library just in case that gets merged, and updated Alby's package to use it. Signet should now parse out-of-the-box now without any additional changes. Let me know if I need to do anything else. Cheers!

rolznz commented on yarn.lock:
```diff
@@ -668,10 +668,10 @@
-  version "3.5.0"
-  resolved "https://registry.yarnpkg.com/@getalby/sdk/-/sdk-3.5.0.tgz#8ef53cda9f22cf31d3e99d3ff4d00366aee0567d"
-  integrity sha512-gnjILgoXOMjCVD1fHcoS3ghhO0UpvFHXuCBb0hz5iuzRYxbPWHXt0eVd60y+z5kR/+T+/S3S2fRMnbCYma+ppQ==
+"@getalby/sdk@^3.4.3":
```
could you undo this change to "@getalby/sdk"?

thebrandonlucas replied:
Done!

rolznz commented:
I tested this with https://signet-app.mutinywallet.com/ to create an invoice and NWC-next using mutinynet configuration connected to the Alby Extension to pay it - looks good!

thebrandonlucas commented:
@rolznz updated!

bumi commented on package.json:
```diff
@@ -52,7 +52,7 @@
     "bech32": "^2.0.0",
     "bitcoinjs-lib": "^6.1.0",
-    "bolt11": "^1.4.1",
+    "bolt11-signet": "^1.4.1",
```
```suggestion
    "bolt11-signet": "1.4.1",
```

bumi reviewed (commented):
would be nice to get it into the official package. until then let's fix the version

thebrandonlucas commented:
> would be nice to get it into the official package. until then let's fix the version

@bumi Agree 100%. If you want to bump here perhaps we can get it in sooner: https://github.com/bitcoinjs/bolt11/pull/73.

Committed your suggestion in the meantime.

bumi commented:
@thebrandonlucas thanks for making the PR there! -and here obviously :)

thebrandonlucas commented:
> @thebrandonlucas thanks for making the PR there! -and here obviously :)

@bumi happy to! let me know if I can do anything else here

rolznz commented on yarn.lock:
```diff
@@ -2865,10 +2865,10 @@ body-parser@1.19.2:
     type-is "~1.6.18"
 
-bolt11@^1.4.1:
+bolt11-signet@^1.4.1:
```
shouldn't this be fixed too? maybe you need to remove and re-install the package?

thebrandonlucas replied:
Unfortunately I need to use this package (my custom fork) until it's merged into the original. I made a PR [here](https://github.com/bitcoinjs/bolt11/pull/73), but I can't do anything but wait until that's merged in. For now, though, there's no reason this fork shouldn't be good to use, since it has everything currently in the original + signet capabilities. As soon as that PR is merged and released I will update it in Alby.

rolznz replied:
I mean, in the yarn.lock file it should be `bolt11-signet@1.4.1:` not `bolt11-signet^1.4.1:`

rolznz replied:
To fix this please remove your package, run yarn install, then re-add it in with the fixed version

thebrandonlucas replied:
Ah, sorry for misunderstanding. Fixed

rolznz reviewed (approved):
tACK

rolznz commented:
Thanks @thebrandonlucas !
